### ISSUE TITLE
actions in redux connect

### ISSUE THREAD

**Issue by lsentkiewicz** [piotrwitek/react-redux-typescript-guide#38]
The component has following props:
```ts
// MyComponent.tsx
export interface MyComponentProps {
  foo: () => any,
  bar: (n: number) => any,
}


const MyComponent: React.SFC<MyComponentProps> = (props) => {
.....
};

```

The container
```ts
// MyComponentContainer.ts
const mapState = (state: State) => ({
  ...
});

export default connect(mapState, {
  foo: actions.foo,
  bar: actions.bar,
})(MyComponent);
```

Few observations:
1. If the action is missing then it doesn't report any error
example: `connect(mapState, {})(MyComponent); // no ts error` 

2. If there is an extra action then it reports an error
example: `connect(mapState, {extra: actions.extra})(MyComponent); // ts error` 

3. Wrong signature (too many arguments).  
The component expects `foo: () => any`, but we connect `(str: string) => any`. No ts error.

4. Wrong signature (missing arguments).
The component expects `bar: (n: number) => any`, but we connect `() => any`. A ts error will be reported.

Is it possible that we receive TS compiler errors in all cases? Especially, 1. case would be useful.

Thanks

**Comment by piotrwitek** [comment]
All cases are by-design TypeScript behaviour, you want to change the behaviour of case 1 and 3.
So:
- Case 1:
This is how redux connect works, if you want a different behaviour you'll need to create custom connect with your own typings.

- Case 3:
You need to enable --strictFunctionTypes compiler option, this will disable bivariant parameter checking for function types (with TS compiler v.2.6+)
